mono/mini/method-to-ir.c

```diff
@@ -7038,8 +7044,7 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				ensure_method_is_allowed_to_call_method (cfg, method, cil_method);
 
 			if (!virtual_ && (cmethod->flags & METHOD_ATTRIBUTE_ABSTRACT))
-				/* MS.NET seems to silently convert this to a callvirt */
-				virtual_ = TRUE;
+				emit_bad_image_failure (cfg, method, cil_method);
```

Comment:
We have this in the interpreter too, and it was specifically introduced for some DIM work: https://github.com/mono/mono/commit/faecfce5d0c29dec56a18f10473a81e52d6acad5#diff-f115ce23fba6528256a47e2b7e8b0a19R1094
We should keep that in sync with the JIT.

The change in the JIT  is quite old, and the original looks like an import 0ab241ce65da

Anyway, looks good to me. Unfortunately it seems to break at least one DIM test.